analysis/Proposed_indicators.md

```diff
@@ -1,6 +1,83 @@
 # Global Indicator Project - proposed indicator discussion
 
 ## 1. Walkability (z-scores for local neigborhood dwelling density, street connectivity and daily living score) 
+#### Carl's thoughts
+If we were calculating these indicators for sample points in the first instance we could use an approach like the following:
+
+In the notebook 'ch_custom_routing_20180824.ipynb' (which you may have accessed from the repository previously, but is just now added back in the ind_bangkok/process/sketches folder) I defined a function to calculate street connectivity for sample points based on their 1600m walkable local neighbourhood.  I believe this approach could be adapated and elaborated on to add in the other measures:
+
+```python
+# Local neighbourhood analysis
+def analyse_local_nh(G, orig_point, buffer=50, length = 1600, intersection_tolerance = 15):
+    # locate closest node on network to 
+    orig_node = ox.get_nearest_node(G, orig_point, return_dist=True)
+    subgraph = nx.ego_graph(G, orig_node[0], radius=length, distance='length')
+    # We project the subgraph to UTM so we can use metres as a unit when buffering
+    subgraph = ox.project_graph(subgraph)
+    su = ox.graph_to_gdfs(subgraph, nodes=False, edges=True, fill_edge_geometry=True)
+    bu = su.geometry.buffer(buffer)
+    uu = bu.geometry.unary_union
+    area_sqm = uu.area
+    area_sqkm = area_sqm*1e-06
+    stats = ox.basic_stats(subgraph, area=area_sqm, clean_intersects=True, tolerance=intersection_tolerance,
+                circuity_dist='euclidean')
+    return({'graph': subgraph,
+             'network_gdfs':su,
+             'buffer':uu,
+             'area_sqkm':area_sqkm,
+             'stats':stats,
+             'origin_node_snap_dist': orig_node[1]})
+     
+# Define some point locations over which we'll iterate
+point_locations = [(-37.808156, 144.964942),(-37.764727, 144.964057)]
+# initialise an 
+nh_estimates = []
+
+for point in point_locations:
+  start = time.time()
+  task = "Calculate local neighbourhood measures for points"
+  nh_estimates.append(analyse_local_nh(G, point))
+  log('Completed task "{}" in {:,.2f} seconds'.format(task,time.time() - start)) 
+  
+# Get some stats
+## area in sqkm
+[x['area_sqkm'] for x in nh_estimates]
+
+## intersection density per sqkm
+[x['stats']['intersection_density_km'] for x in nh_estimates]
+
+## clean intersection density per sqkm
+[x['stats']['clean_intersection_density_km'] for x in nh_estimates]
+```
+So, that makes use of the osmnx stats functionality for calculating street connectivity.  
+
+The use of the function might be overkill; assuming we have a dataframe of sample points, if you coded it right there might even be a vectorised solution to carry out the operations, like extracting the 50m buffered unary union of 1600m network linestrings for a sample point as a geometry field and then use intersection analysis for each sample point's network buffer polygon with other pre-prepared resources:
```

Comment:
Thanks, Carl. We will adopt this function to get the local neighbourhood and stats.  Yes, assumptions have to be made to this, but it is ok as far as we can get it working consistently across the global context. Regarding your sample point question, we went back to your [original post](https://3.basecamp.com/3662734/buckets/11779922/messages/1770544113) on basecamp, and thought it may be good to use the regular sampling approach as you proposed. Geoff has proposed using OSMnx functions for vertices distribution, I will test on this later this week. As long as we get this working, we could weight by population to get the weighted-sampling point.

Reply:
That sounds great!  There was some code I pulled from one of OSMnx's other functions using the redistribute_verices() function earlier on in the post you linked to (second comment).  However, I'm sure between yourself and Geoff you will have implemented something concrete using this --- looking forward to checking out the code!